### PR TITLE
docs: convert useListenFocusTriggers stories file to TS

### LINKED ISSUE #2474: docs: convert useListenFocusTriggers stories file to TS
Convert the `useListenFocusTriggers` stories file to TypeScript:
- Change the file extension to `stories.tsx`
- Resolve type issues (if there are), avoid using `any` and prefer existing types

## Related files
`packages/core/src/hooks/useListenFocusTriggers/__stories__/useListenFocusTriggers.stories.js`

---

[Monday ticket [internal]](https://monday.monday.com/boards/3532714909/pulses/7601008784)

### PR DESCRIPTION
**Pull request by LoayTarek5**
Here are the main changes made to convert the code to TypeScript:
Added type annotation for the text state: useState<string>("-").
Specified the type for the ref: useRef<HTMLButtonElement>(null).
Added type annotations for the event parameters in onFocusByMouse and onFocusByKeyboard: React.FocusEvent<HTMLButtonElement>.
Added an empty interface RenderProps which can be extended later if needed.
Added as const to the Overview object to ensure type inference for its properties.
Resolves #2474 
- [x] I have read the [Contribution Guide](../packages/core/CONTRIBUTING.md) for this project.

### CODE REVIEW

**Comment by LoayTarek5** [comment]
@LastExileUA @virtser @doronbrikman @jasonbahl 
could some one look at it😃

**Review by talkor — CHANGES_REQUESTED** [review]
@LoayTarek5 thank you for this PR! Looks like there are two similar files and the previous one is missing in this changeset (which means it's not deleted). Let's add the deletion of the old file, and keep only the tsx file here
Thanks!

**Comment by talkor** [comment]
@LoayTarek5 there is still the old file :(

File: packages/core/src/hooks/useListenFocusTriggers/__stories__/useListenFocusTriggers.stories.tsx [hunk 1]
```diff
@@ -4,19 +4,21 @@ import { Button, Flex } from "../../../components";
 import "../../__stories__/general-hooks-stories.scss";
 
 export default {
-  title: "Hooks/useListenFocusTriggers"
+  title: "Hooks/useListenFocusTriggers"   
 };
 
+interface RenderProps {}
```
Comment (talkor) [inline packages/core/src/hooks/useListenFocusTriggers/__stories__/useListenFocusTriggers.stories.tsx:10]: Looks like it's unused :)

File: packages/core/src/hooks/useListenFocusTriggers/__stories__/useListenFocusTriggers.stories.tsx [hunk 2]
```diff
@@ -35,4 +37,4 @@ export const Overview = {
   },
 
   name: "Overview"
-};
+} as const;
```
Comment (talkor) [inline packages/core/src/hooks/useListenFocusTriggers/__stories__/useListenFocusTriggers.stories.tsx:40]: I don't think it's needed

**Review by talkor — COMMENTED** [review]
Great @LoayTarek5 ! Just check out my 2 comments, probably unused code

**Review by talkor — APPROVED** [review]
Thank you @LoayTarek5 ! Looks good, please just run prettier on this file as the CI fails for the extra spaces in line 7 
Thanks!

**Pull request merged**